hoptimator-util/src/main/java/com/linkedin/hoptimator/util/planner/PipelineRel.java

```diff
@@ -78,7 +82,22 @@ public void setSink(String database, List<String> path, RelDataType rowType, Map
       this.sinkDatabase = database;
       this.sinkPath = path;
       this.sinkRowType = rowType;
-      this.sinkOptions = options;
+      this.sinkOptions = addKeysAsOption(options, rowType);
+    }
+
+    private Map<String, String> addKeysAsOption(Map<String, String> options, RelDataType rowType) {
```

Comment:
I don't love this approach, very brute force.

I looked into hints to solve this and did get them working to an extent (will open a separate PR) but this would require users to pass in key information into their SQL statement. I have not figure out a way to inject hints at runtime from VeniceDriver.

Reply:
I guess I'm surprised we need to fully specify the keys in the options. The Kafka connector has similar properties (`key.prefix`, `key.fields`), but you don't need both. Is the Venice connector doing something different here? I'd expect `key.prefix=key_` to be sufficient.

Reply:
Also, how would the Venice connector behave if we grouped the keys in a `Row(...)` object? Can we just have `key.fields=KEY` and then `KEY ROW(F1 VARCHAR, F2 INT)` etc?

(Not suggesting we do that, just asking if possible?)

Reply:
> I guess I'm surprised we need to fully specify the keys in the options. The Kafka connector has similar properties (`key.prefix`, `key.fields`), but you don't need both. Is the Venice connector doing something different here? I'd expect `key.prefix=key_` to be sufficient.

Yea I confirmed it is an issue with Venice due do some additional avro schema validation they do. `key.prefix` and `key.fields` do seem to be used differently and need to be used together.

Reply:
Looking into the ROW syntax and it doesn't seem that is possible in Flink, there is no way to get Flink to destruct that ROW

Reply:
Makes sense. I think that's fine. The only slight concern I have is potential deviation from the Kafka -> Kafka use-case. With Kafka -> Kafka users can do `select *` and retain partitioning, since the output topic will use the input topic's `KEY` field. Users might expect `select *` to work similarly with Kafka -> Venice, except Kafka has one key (`KEY`) and Venice has one or more keys (`KEY_xyz`), so that might not work. I'm thinking we should actually change the way Kafka works and adopt this `KEY_xzy` approach, or maybe have `PipelineRel` explicitly use `KEY` in cases where there is only one key.

Reply:
What does retaining partitioning mean for the Kafka -> Venice use case? It seems like it is more of a problem on the producer side. Users that expect the same partitioning behavior would have to key their Kafka topic using the same combination of keys as Venice. We did this in Brooklin by constructing the producer key as a simple string with key values separated by `_` from the source keys. Of course this isn't the same as identity partitioning but it does ensure that downstream consumer tasks read the same combo of keys.

Even for the Kafka -> Kafka use case, we aren't the ones consuming, the partitioning behavior comes from Flink right? I haven't looked into it to be fair, not sure how the behavior changes if you define `key.fields` there.